Sources/Flows/OAuth2.swift

```diff
@@ -366,9 +366,14 @@ open class OAuth2: OAuth2Base {
 				do {
 					let data = try response.responseData()
 					let json = try self.parseRefreshTokenResponseData(data)
-					if response.response.statusCode >= 400 {
+					switch response.response.statusCode {
+					case 400..<500:
```

Comment:
That is a good point. Yes, it would probably be good to add `OAuth2Error.clientError(Int)`, which takes the status code, and change `OAuth2Error.serverError` to `OAuth2Error.serverError(Int)` and raise either of these two if status >= 400 (and never delete the token). @catalinaturlea , what do you think?

Reply:
Just to be on the safe side, @p2 could you confirm that this is ok not deleting the token here? I am just worried if nil refresh tokens are not used to check some state somewhere else in the library.